xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.h

```diff
@@ -42,6 +42,31 @@ struct VideoPicture
   VideoPicture& SetParams(const VideoPicture &pic);
   void Reset(); // reinitialize members, videoBuffer will be released if set!
 
+  /*!
+  * \brief Checks if the video picture has A53 side data (closed captions).
+  *
+  * \return true if the video picture has A53 side data, false otherwise
+  */
+  bool HasA53SideData() const { return m_A53SideData != nullptr; }
+
+  /*!
+  * \brief Gets the A53 side data of the video picture (closed captions)
+  * \return the A53 side data, reseting the value
+  */
+  AVBufferRef* ConsumeA53SideData()
```

Comment:
Shoud this be `GetA53SideData` ? As you say that in the doxy.

Reply:
It might also make sense to have a `ReleaseA53SideData` method that calls the `av_buffer_unref` that way the ref and unref are contained.

Reply:
I opted for the consume Vs get because it's set as nullptr. It'd be better to just use smart pointers here and move the pointer but I couldn't manage to do it

Reply:
@lrusak if you are bored enough and could help with this it'd be appreciated. I'd say the proper way to do this is to store a unique_ptr for the side data in this class instead of a raw pointer and move it when calling get, so that it's set to nullptr internally in the class. Moreover I think it could in theroty have a custom deleter that would unref the buffer on destruction, avoiding the callee to call into unref manually.
Unfortunately I couldn't manage to do it because of the default copy constructors we have on VideoPicture if I recall correctly.